From 7c3cd334f6f25360b06c648a20efab2788c55a09 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Sun, 04 Oct 2020 15:59:13 +0300
Subject: [PATCH] [SW-14663] ppboot: update descriptor verify

Previously, descriptor verification logic did not allow using older
descriptor version. But since image descriptor is backward compatible,
the logic is changed so that older descriptor version number is
allowed.

Change-Id: Ieb684f5c430e3c72291a744ab4f0e33da4553d4d
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/ppboot/boot_desc.c b/ppboot/boot_desc.c
index 0bfc9a6..8c07667 100644
--- a/ppboot/boot_desc.c
+++ b/ppboot/boot_desc.c
@@ -43,7 +43,7 @@
 	if ((img_desc->magic.val & 0xFFFFFF00) != HL_FW_IMG_DESC_MAGIC)
 		return false;
 
-	if (img_desc->desc_ver > HL_FW_IMG_DESC_CUR_VER)
+	if (img_desc->desc_ver == 0 || img_desc->desc_ver == ~0)
 		return false;
 
 	if (img_desc->idx >= 2)
